plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/3dtiles/BuildingFilter/index.tsx

```diff
@@ -0,0 +1,112 @@
+import { Slider } from "@web/sharedComponents";
+import { styled } from "@web/theme";
+import { CSSProperties, useMemo } from "react";
+
+import { BaseFieldProps } from "../../types";
+
+import useHooks from "./hooks";
+import { MAX_ABOVEGROUND_FLOOR, MAX_BASEMENT_FLOOR, MAX_HEIGHT } from "./useBuildingFilter";
+
+const rangeToText = (range: [from: number, to: number]) => range.join(" ~ ");
+
+const BuildingFilter: React.FC<BaseFieldProps<"buildingFilter">> = ({
+  value,
+  dataID,
+  editMode,
+  onUpdate,
+}) => {
+  const { options, handleUpdateRange } = useHooks({
+    value,
+    dataID,
+    onUpdate,
+  });
+
+  const styleProps = useMemo(
+    () => ({
+      trackStyle: [
+        {
+          backgroundColor: "#00BEBE",
+        },
+      ] as CSSProperties[],
+      handleStyle: [
+        {
+          border: "2px solid #00BEBE",
+        },
+        {
+          border: "2px solid #00BEBE",
+        },
+      ] as CSSProperties[],
+    }),
+    [],
+  );
```

Comment:
Can't this just be moved outside the component instead of in a useMemo?

Reply:
I fixed.
[fix: cull out styleProps to out of component](https://github.com/eukarya-inc/reearth-plateauview/pull/121/commits/5746fd20476a11ce24396a9674ca2c61198e51a3)